pkg/local_object_storage/shard/evacuate.go

```diff
@@ -0,0 +1,106 @@
+package shard
+
+import (
+	"encoding/binary"
+	"errors"
+	"os"
+
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor"
+)
+
+var dumpMagic = []byte("NEOF")
+
+// EvacuatePrm groups the parameters of Evacuate operation.
+type EvacuatePrm struct {
+	path string
+}
+
+// WithPath is an Evacuate option to set the destination path.
+//
+// Option is required.
+func (p *EvacuatePrm) WithPath(path string) *EvacuatePrm {
+	p.path = path
+	return p
+}
+
+// EvacuateRes groups the result fields of Evacuate operation.
+type EvacuateRes struct {
+	count int
+}
+
+// Count return amount of object written.
+func (r *EvacuateRes) Count() int {
+	return r.count
+}
+
+var ErrMustBeReadOnly = errors.New("shard must be in read-only mode")
+
+// Evacuate dumps all objects from the shard to a file.
+//
+// Returns any error encountered.
+func (s *Shard) Evacuate(prm *EvacuatePrm) (*EvacuateRes, error) {
+	s.m.RLock()
+	defer s.m.RUnlock()
+
+	if s.info.Mode != ModeReadOnly {
+		return nil, ErrMustBeReadOnly
+	}
+
+	f, err := os.Create(prm.path)
```

Comment:
do we need the dump file to have any specific permissions? @realloc @alexvanin

Reply:
`0640` would be enough.
There should be an option to stream the dump, let  us don't forget to implement it later.

Reply:
Changed mode, added streams for evacuate/restore.